expertsystem/ui/__init__.py

```diff
@@ -237,20 +239,33 @@ def _create_seed_graphs(
 
     @staticmethod
     def _convert_edges_to_dict(
-        graphs: List[StateTransitionGraph[ParticleWithSpin]],
-    ) -> List[StateTransitionGraph[dict]]:
-        for graph in graphs:
-            for edge_id, edge_property in graph.edge_props.items():
-                if not isinstance(edge_property, dict):
-                    graph.edge_props[
-                        edge_id
-                    ] = particle_with_spin_projection_to_dict(  # type: ignore
-                        edge_property
-                    )
-        return graphs  # type: ignore
+        instance: Union[
+            List[StateTransitionGraph[ParticleWithSpin]],
+            List[Tuple[StateTransitionGraph, NodeSettings]],
+            GraphSettingsGroups,
+        ],
+    ) -> None:
+        # https://github.com/ComPWA/expertsystem/issues/254
+        def convert_edges_in_graph(
+            graphs: List[StateTransitionGraph[ParticleWithSpin]],
+        ) -> None:
+            for graph in graphs:
+                for edge_id, edge_property in graph.edge_props.items():
+                    if not isinstance(edge_property, dict):
+                        graph.edge_props[
+                            edge_id
+                        ] = particle_with_spin_projection_to_dict(  # type: ignore
+                            edge_property
+                        )
+
+        if isinstance(instance, list) and len(instance) > 0:
+            if isinstance(instance[0], StateTransitionGraph):
+                convert_edges_in_graph(instance)  # type: ignore
+            elif isinstance(instance[0], tuple):
+                convert_edges_in_graph([g for g, _ in instance])  # type: ignore
```

Comment:
does this work? since a tuple is immutable it should be not possible to alter it in place

Reply:
Yes, it's a `tuple` of two, so you unpack the first into `g` and the second into `_`